crates/swc/tests/errors/lints/no-alert/alert-ident-hoisting/output.swc-stderr

```diff
@@ -0,0 +1,6 @@
+error: Unexpected alert
+ 
+  |
+1 | alert();
```

Comment:
I think this should not be an error, as user is calling user-defined `alert`.
If we want to report an error, it should be usage before declarationn.

Reply:
you're mean is should be caught by another rule?

Reply:
I meant this should not be treated as a call to the global `alert`.

Reply:
Ok. I'll fix it

Reply:
@kdy1 while I investigating how to solve your comment I did found different syntax context resolving behaviour in tests in swc and swc_ecma_lint crates. I managed to solve different behaviour by adding to `crates/swc/src/config/mod.rs` following command 

```
program.visit_mut_with(&mut resolver_with_mark(top_level_mark));
```

It is correct solution or it should be fixed with another way?

Reply:
It's wrong and I think we need a feature gate for non-critical errors.

Reply:
So I need help with understanding how to fix problem with different syntax contexts resolving 😃

Reply:
Ah, are you using collecting bindings on creating?

Reply:
I think you can do it in `lint_module` instead?
If it does not work, it needs some thinking about desgin...

Reply:
Oh, I think your fix is correct way to go (applying resolver), although we need to remove the next call to the resolver.

Reply:
Okay. Happy to hear it

Reply:
hm... My fix solves my problem, but it fail some tests 

```
    tests_tests__fixture__issue_1160__input
    tests_tests__fixture__issue_1345__input
    tests_tests__fixture__issue_1362__case1__input
    tests_tests__fixture__issue_1421__case1__input
    tests_tests__fixture__issue_1869__decorator__input
    tests_tests__fixture__issue_846__case1__input
 ```

Reply:
I can commit my fixes or provide terminal output

Reply:
I'll create another PR to fix the wrong design and you can rebase after it's merged.

Reply:
I made https://github.com/swc-project/swc/pull/3400

Reply:
Thank you